Src/AutoFixture/Kernel/TypeEnvy.cs

```diff
@@ -11,17 +11,12 @@ internal static class TypeEnvy
         public static TAttribute GetAttribute<TAttribute>(object candidate)
             where TAttribute : Attribute
         {
-            var member = candidate as MemberInfo;
-            if(member != null)
-            {
-                return member.GetCustomAttribute<TAttribute>();
-            }
-            var parameter = candidate as ParameterInfo;
-            if(parameter != null)
-            {
-                return parameter.GetCustomAttribute<TAttribute>();
-            }
-            return null;
+            var attributeProvider = candidate as ICustomAttributeProvider;
+
+            return attributeProvider?
+                .GetCustomAttributes(typeof(TAttribute), true)
+                .Cast<TAttribute>()
```

Comment:
There were a number of core classes that were used a lot on the 'hot path'. Back in 
#218 we removed some of the LINQ statements in favour of less allocations. I'm not requesting that you should change this, but I thought it's worthwhile briefly considering if it might have any impact.

We left a few comments in various key places like [this one]( https://github.com/AutoFixture/AutoFixture/pull/218/commits/0d9d95f1e4ea0008a7f68d46a67756cd37035d52).